lib/operations/cursor_ops.js

```diff
@@ -53,6 +52,11 @@ function count(cursor, applySkipLimit, opts, callback) {
   if (opts.limit) command.limit = opts.limit;
   if (cursor.s.options.hint) command.hint = cursor.s.options.hint;
 
+  // Ensure we have the right read preference inheritance
+  if (opts.readPreference) {
```

Comment:
It looks like we should either attempt to merge `count` from `collection_ops` with this, or at least be doing the same things in both places (option #2 might be a good place to start when approaching option #1). Off the top it looks like we should be using `decorateWithReadConcern` and `resolveReadPreference`